### ISSUE TITLE
PortManager crashing when using -Djava.net.preferIPv4Stack=true

### ISSUE THREAD

**Issue by mathieucarbou** [Terracotta-OSS/terracotta-utilities#31]
Crashes when running JVM with `-Djava.net.preferIPv4Stack=true`:

```
java.lang.RuntimeException: java.lang.IllegalArgumentException: java.net.UnknownHostException: no such interface en6
	at org.terracotta.utilities.test.net.NetStat.info(NetStat.java:93)
	at org.terracotta.utilities.test.net.PortManager.diagnosticReleaseCheck(PortManager.java:511)
	at org.terracotta.utilities.test.net.PortManager.lambda$combine$5(PortManager.java:666)
	at org.terracotta.utilities.test.net.PortManager$PortRef.lambda$close$2(PortManager.java:650)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.terracotta.utilities.test.net.PortManager$PortRef.close(PortManager.java:650)
	at org.terracotta.angela.common.net.DefaultPortAllocator.close(DefaultPortAllocator.java:71)
	at net.schmizz.sshj.common.IOUtils.closeQuietly(IOUtils.java:41)
	at net.schmizz.sshj.common.IOUtils.closeQuietly(IOUtils.java:29)
	at org.terracotta.angela.BaseIT.close(BaseIT.java:117)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.lang.IllegalArgumentException: java.net.UnknownHostException: no such interface en6
	at org.terracotta.utilities.test.net.NetStat$BusyPort$IPVersion.getInetSocketAddress(NetStat.java:1124)
	at org.terracotta.utilities.test.net.NetStat$BusyPort$Builder.processEndpoint(NetStat.java:1042)
	at org.terracotta.utilities.test.net.NetStat$BusyPort$Builder.localEndpoint(NetStat.java:1011)
	at org.terracotta.utilities.test.net.NetStat.processNettopEndpoints(NetStat.java:710)
	at org.terracotta.utilities.test.net.NetStat.parseNetTop(NetStat.java:659)
	at org.terracotta.utilities.test.net.NetStat.access$600(NetStat.java:70)
	at org.terracotta.utilities.test.net.NetStat$Platform$2.lambda$netstat$0(NetStat.java:143)
	at org.terracotta.utilities.test.net.NetStat$Platform.runCommand(NetStat.java:274)
	at org.terracotta.utilities.test.net.NetStat$Platform.access$200(NetStat.java:110)
	at org.terracotta.utilities.test.net.NetStat$Platform$2.netstat(NetStat.java:143)
	at org.terracotta.utilities.test.net.NetStat.info(NetStat.java:91)
	... 41 common frames omitted
Caused by: java.net.UnknownHostException: no such interface en6
	at java.base/java.net.Inet6Address.initstr(Inet6Address.java:488)
	at java.base/java.net.Inet6Address.<init>(Inet6Address.java:409)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1357)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
	at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
	at org.terracotta.utilities.test.net.NetStat$BusyPort$IPVersion.getInetSocketAddress(NetStat.java:1120)
	... 51 common frames omitted
```

This is because the output of `nettop` (macosx) contains a mix of ipv4 / ipv6 stuff and the jVM does not understand IPv6

```
> nettop -L1 -m tcp -n -J state
[...]
tcp6 ::1.49693<->*.*,Listen,
tcp4 127.0.0.1:49693<->*:*,Listen,
[...]
```

**Comment by mathieucarbou** [comment]
```bash
/Library/Java/JavaVirtualMachines/11/Contents/Home/bin/java -Djava.net.preferIPv4Stack=true -classpath [...] org.terracotta.utilities.test.net.NetStat
```

output:

```
13:55:15.406 [main] TRACE o.t.u.t.n.NetStat:271 - Command complete [nettop, -L1, -m, tcp, -n, -J, state]; rc=0
    ,state,
    remoted.102,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49152<->fe80::aede:48ff:fe33:4455%en6.59602,Established,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49153<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49154<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49155<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49156<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49157<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49158<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49159<->*.*,Listen,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49160<->*.*,Listen,
    apsd.126,,
    tcp4 192.168.125.134:52747<->17.57.146.163:5223,Established,
    syspolicyd.160,,
    tcp4 192.168.125.168:52165<->17.248.145.105:443,Established,
    bosUpdateProxy.275,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.50375<->fe80::aede:48ff:fe33:4455%en6.49227,Established,
    corekdld.276,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49161<->fe80::aede:48ff:fe33:4455%en6.49229,Established,
    SubmitDiagInfo.277,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49153<->fe80::aede:48ff:fe33:4455%en6.49237,Established,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49170<->fe80::aede:48ff:fe33:4455%en6.49235,Established,
    findmydeviced.306,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49162<->fe80::aede:48ff:fe33:4455%en6.49225,Established,
    biometrickitd.309,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.57696<->fe80::aede:48ff:fe33:4455%en6.49224,Established,
    rapportd.431,,
    tcp4 *:58040<->*:*,Listen,
    tcp6 *.58040<->*.*,Listen,
    tcp4 192.168.125.168:58040<->192.168.125.138:49968,Established,
    mobileactivatio.434,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.49183<->fe80::aede:48ff:fe33:4455%en6.49232,Established,
    Microsoft To Do.457,,
    tcp4 192.168.125.168:51997<->40.90.136.182:443,Established,
    tcp4 192.168.125.168:51991<->40.101.91.82:443,Established,
    tcp4 192.168.125.168:58066<->52.98.206.210:443,Established,
    tcp4 192.168.125.168:63277<->40.101.11.178:443,Established,
    ControlCenter.472,,
    tcp4 *:7000<->*:*,Listen,
    tcp6 *.7000<->*.*,Listen,
    tcp4 *:5000<->*:*,Listen,
    tcp6 *.5000<->*.*,Listen,
    Google Chrome H.685,,
    tcp4 192.168.125.168:52240<->3.66.141.242:443,Established,
    tcp4 192.168.125.168:52183<->20.71.240.215:443,SynSent,
    tcp4 192.168.125.168:52181<->20.71.240.215:443,SynSent,
    tcp4 192.168.125.168:58042<->192.168.125.112:8009,Established,
    tcp4 192.168.125.168:58043<->192.168.125.117:8009,Established,
    tcp4 192.168.125.168:58044<->192.168.125.176:8009,Established,
    tcp4 192.168.125.168:58045<->192.168.125.108:42886,Established,
    tcp4 192.168.125.168:58046<->192.168.125.108:8009,Established,
    tcp4 192.168.125.168:58047<->192.168.125.119:8009,Established,
    tcp4 192.168.125.168:58048<->192.168.125.110:8009,Established,
    tcp4 192.168.125.168:58052<->192.168.125.107:8009,Established,
    tcp4 192.168.125.168:58053<->192.168.125.113:8009,Established,
    tcp4 192.168.125.168:58122<->185.45.152.28:443,Established,
    tcp4 192.168.125.168:58253<->54.86.238.201:443,Established,
    tcp4 192.168.125.168:58473<->52.205.53.167:443,Established,
    tcp4 192.168.125.168:60145<->3.82.237.223:443,Established,
    tcp4 192.168.125.168:60674<->34.255.245.51:443,Established,
    tcp4 192.168.125.168:61015<->179.60.192.23:443,Established,
    tcp4 192.168.125.168:61275<->35.174.127.31:443,Established,
    tcp4 192.168.125.168:61277<->74.125.133.188:5228,Established,
    tcp4 192.168.125.168:61278<->192.168.125.110:8008,Established,
    tcp4 192.168.125.168:61279<->192.168.125.107:8008,Established,
    tcp4 192.168.125.168:61281<->18.66.192.12:443,Established,
    tcp4 192.168.125.168:61283<->142.250.184.197:443,Established,
    tcp4 192.168.125.168:61293<->20.189.173.3:443,Established,
    tcp4 192.168.125.168:61299<->31.13.84.51:443,Established,
    tcp4 192.168.125.168:63842<->52.182.141.63:443,Established,
    tcp4 192.168.125.168:49959<->185.199.111.133:443,Established,
    tcp4 192.168.125.168:49964<->185.199.108.154:443,Established,
    tcp4 192.168.125.168:50130<->54.239.37.226:443,Established,
    tcp4 192.168.125.168:50131<->54.239.37.226:443,Established,
    tcp4 192.168.125.168:50132<->54.239.37.226:443,Established,
    tcp4 192.168.125.168:50229<->140.82.114.25:443,Established,
    corespeechd.868,,
    tcp6 fe80::aede:48ff:fe00:1122%en6.58054<->fe80::aede:48ff:fe33:4455%en6.49213,Established,
    pCloud Drive.905,,
    tcp4 127.0.0.1:8989<->*:*,Listen,
    tcp4 *:49461<->*:*,Listen,
    tcp4 192.168.125.168:58105<->195.154.58.244:443,Established,
    Loupedeck.907,,
    tcp4 *:49692<->*:*,Listen,
    tcp4 *:49694<->*:*,Listen,
    tcp4 *:49695<->*:*,Listen,
    tcp6 ::1.49693<->*.*,Listen,
    tcp4 127.0.0.1:49693<->*:*,Listen,
    Adobe Desktop S.982,,
    tcp4 127.0.0.1:15292<->*:*,Listen,
    tcp4 127.0.0.1:15393<->*:*,Listen,
    tcp4 127.0.0.1:16494<->*:*,Listen,
    Adobe CEF Helpe.985,,
    tcp4 127.0.0.1:57294<->127.0.0.1:49890,Established,
    node.1022,,
    tcp4 127.0.0.1:49890<->*:*,Listen,
    tcp4 127.0.0.1:49891<->*:*,Listen,
    tcp4 127.0.0.1:45623<->*:*,Listen,
    tcp4 127.0.0.1:49530<->*:*,Listen,
    tcp4 127.0.0.1:49890<->127.0.0.1:57294,Established,
    Core Sync.1038,,
    tcp4 192.168.125.168:58286<->100.25.95.143:443,Established,
    tcp4 192.168.125.168:58390<->18.235.195.121:443,Established,
    idea.35401,,
    tcp4 127.0.0.1:61161<->127.0.0.1:52484,Established,
    tcp4 127.0.0.1:52482<->127.0.0.1:52483,Established,
    tcp4 127.0.0.1:52482<->*:*,Listen,
    tcp4 127.0.0.1:51737<->127.0.0.1:61193,Established,
    tcp4 127.0.0.1:51736<->127.0.0.1:61219,Established,
    tcp4 127.0.0.1:6942<->*:*,Listen,
    tcp4 127.0.0.1:63342<->*:*,Listen,
    tcp4 127.0.0.1:61161<->*:*,Listen,
    tcp4 127.0.0.1:61194<->127.0.0.1:41359,Established,
    tcp4 127.0.0.1:61220<->127.0.0.1:38816,Established,
    tcp4 127.0.0.1:51187<->127.0.0.1:61219,Established,
    tcp4 127.0.0.1:51382<->127.0.0.1:61193,Established,
    java.35480,,
    tcp4 127.0.0.1:61193<->127.0.0.1:51737,Established,
    tcp4 127.0.0.1:61193<->127.0.0.1:51382,Established,
    tcp4 127.0.0.1:41359<->*:*,Listen,
    tcp4 127.0.0.1:61193<->*:*,Listen,
    tcp4 127.0.0.1:41359<->127.0.0.1:61194,Established,
    java.35619,,
    tcp4 127.0.0.1:61219<->127.0.0.1:51736,Established,
    tcp4 127.0.0.1:38816<->*:*,Listen,
    tcp4 127.0.0.1:61219<->*:*,Listen,
    tcp4 127.0.0.1:38816<->127.0.0.1:61220,Established,
    tcp4 127.0.0.1:61219<->127.0.0.1:51187,Established,
    PacketTunnel.35706,,
    tcp4 192.168.125.168:61276<->193.26.194.88:691,Established,
    Authy Desktop.35708,,
    tcp4 192.168.125.168:61245<->34.194.101.32:443,Established,
    Mattermost Help.36234,,
    tcp4 10.248.15.59:52233<->157.189.192.72:443,Established,
    tcp4 10.248.15.59:52232<->157.189.192.72:443,Established,
    tcp4 10.248.15.59:64434<->157.189.192.72:443,Established,
    tcp4 192.168.125.168:64448<->185.199.111.154:443,Established,
    tcp4 10.248.15.59:51224<->157.189.192.72:443,Established,
    java.36556,,
    tcp4 127.0.0.1:10988<->127.0.0.1:52252,Established,
    tcp4 127.0.0.1:52252<->127.0.0.1:10988,Established,
    tcp6 *.10988<->*.*,Listen,
    tcp4 127.0.0.1:10988<->127.0.0.1:52242,Established,
    tcp4 127.0.0.1:52242<->127.0.0.1:10988,Established,
    tcp6 *.25343<->*.*,Listen,
    tcp6 *.12552<->*.*,Listen,
    java.36564,,
    tcp6 *.4874<->*.*,Listen,
    tcp6 *.25079<->*.*,Listen,
    java.36576,,
    tcp6 *.40380<->*.*,Listen,
    tcp6 *.10056<->*.*,Listen,
    tcp4 127.0.0.1:4415<->127.0.0.1:52474,Established,
    tcp4 127.0.0.1:52474<->127.0.0.1:4415,Established,
    tcp6 *.19664<->*.*,Listen,
    tcp6 *.4415<->*.*,Listen,
    java.36619,,
    tcp6 *.42862<->*.*,Listen,
    tcp6 *.20694<->*.*,Listen,
    java.36623,,
    tcp4 127.0.0.1:52484<->127.0.0.1:61161,Established,
    java.36624,,
    tcp4 127.0.0.1:52483<->127.0.0.1:52482,Established,
Exception in thread "main" java.lang.RuntimeException: java.lang.IllegalArgumentException: java.net.UnknownHostException: no such interface en6
	at org.terracotta.utilities.test.net.NetStat.info(NetStat.java:93)
	at org.terracotta.utilities.test.net.NetStat.main(NetStat.java:102)
Caused by: java.lang.IllegalArgumentException: java.net.UnknownHostException: no such interface en6
	at org.terracotta.utilities.test.net.NetStat$BusyPort$IPVersion.getInetSocketAddress(NetStat.java:1124)
	at org.terracotta.utilities.test.net.NetStat$BusyPort$Builder.processEndpoint(NetStat.java:1042)
	at org.terracotta.utilities.test.net.NetStat$BusyPort$Builder.localEndpoint(NetStat.java:1011)
	at org.terracotta.utilities.test.net.NetStat.processNettopEndpoints(NetStat.java:710)
	at org.terracotta.utilities.test.net.NetStat.parseNetTop(NetStat.java:659)
	at org.terracotta.utilities.test.net.NetStat.access$600(NetStat.java:70)
	at org.terracotta.utilities.test.net.NetStat$Platform$2.lambda$netstat$0(NetStat.java:143)
	at org.terracotta.utilities.test.net.NetStat$Platform.runCommand(NetStat.java:274)
	at org.terracotta.utilities.test.net.NetStat$Platform.access$200(NetStat.java:110)
	at org.terracotta.utilities.test.net.NetStat$Platform$2.netstat(NetStat.java:143)
	at org.terracotta.utilities.test.net.NetStat.info(NetStat.java:91)
	... 1 more
Caused by: java.net.UnknownHostException: no such interface en6
	at java.base/java.net.Inet6Address.initstr(Inet6Address.java:488)
	at java.base/java.net.Inet6Address.<init>(Inet6Address.java:409)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1357)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
	at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
	at org.terracotta.utilities.test.net.NetStat$BusyPort$IPVersion.getInetSocketAddress(NetStat.java:1120)
	... 11 more
```

**Comment by mathieucarbou** [comment]
> ```shell
> /Library/Java/JavaVirtualMachines/11/Contents/Home/bin/java
> ```

Same issue with both 8 and 11:

```
❯  /Library/Java/JavaVirtualMachines/11/Contents/Home/bin/java -version
openjdk version "11.0.10" 2021-01-19 LTS
OpenJDK Runtime Environment Zulu11.45+27-CA (build 11.0.10+9-LTS)
OpenJDK 64-Bit Server VM Zulu11.45+27-CA (build 11.0.10+9-LTS, mixed mode)
```

```
❯  /Library/Java/JavaVirtualMachines/8/Contents/Home/bin/java -version
openjdk version "1.8.0_282"
OpenJDK Runtime Environment (Zulu 8.52.0.23-CA-macosx) (build 1.8.0_282-b08)
OpenJDK 64-Bit Server VM (Zulu 8.52.0.23-CA-macosx) (build 25.282-b08, mixed mode)
```

**Comment by cljohnso** [comment]
Closed by #32
